src/node/snapshotter.h

```diff
@@ -217,6 +225,41 @@ namespace ccf
       return false;
     }
 
+    void record_signature(
+      consensus::Index idx,
+      const std::vector<uint8_t>& sig,
+      const NodeId& node_id)
+    {
+      std::lock_guard<std::mutex> guard(lock);
+
+      for (auto& pending_snapshot : pending_snapshots)
+      {
+        if (
+          pending_snapshot.evidence_idx < idx &&
+          !pending_snapshot.sig.has_value())
+        {
+          pending_snapshot.node_id = node_id;
+          pending_snapshot.sig = sig;
+        }
+      }
+    }
+
+    void record_serialised_tree(
+      consensus::Index idx, const std::vector<uint8_t>& tree)
+    {
+      std::lock_guard<std::mutex> guard(lock);
+
+      for (auto& pending_snapshot : pending_snapshots)
+      {
+        if (
+          pending_snapshot.evidence_idx < idx &&
+          !pending_snapshot.tree.has_value())
+        {
+          pending_snapshot.tree = tree;
+        }
+      }
+    }
```

Comment:
Through these two functions that are called from a local hook on the signature and serialised tree tables, the snaphotter records the signature and proof for yet-to-be-committed snapshots.